plugins/inputs/modbus/request.go

```diff
@@ -8,9 +8,25 @@ type request struct {
 	address uint16
 	length  uint16
 	fields  []field
+	tags    map[string]string
 }
 
-func newRequestsFromFields(fields []field, maxBatchSize uint16) []request {
+func newRequest(f field, tags map[string]string) request {
+	r := request{
+		address: f.address,
+		length:  f.length,
+		fields:  []field{f},
+		tags:    map[string]string{},
+	}
+
+	// Copy the tags
+	for k, v := range tags {
+		r.tags[k] = v
+	}
+	return r
+}
+
+func newRequestsFromFields(fields []field, maxBatchSize uint16, tags map[string]string) []request {
```

Comment:
Since this now takes tags too, let's rename the func to newRequests. It might also be clearer if the arguments are ordered fields, tags, maxBatchSize.

Reply:
Done.